### PR TITLE
Bulk Data import job with pod restarts unable to delete #3033

### PR DESCRIPTION
**Pull request by prb112**
- Added a missing variable defaultHostName (we used internally however did not move to opensource)
 - added batch db hostname variable for consistency 

Signed-off-by: Paul Bastide <pbastide@us.ibm.com>

### CODE REVIEW

File: docs/src/pages/guides/FHIRServerUsersGuide.md [hunk 1]
```diff
@@ -1488,6 +1488,8 @@ Note: If you use PostgreSQL database as IBM FHIR Server data store or the JavaBa
 
 For more information about Liberty JavaBatch configuration, please refer to [IBM WebSphere Liberty Java Batch White paper](https://www-03.ibm.com/support/techdocs/atsmastr.nsf/webindex/wp102544).
 
+If you are running in a Kubernetes deployment, be sure to set the environment variable MY_POD_NAME to `metadata.name` as shown in [deployment.yaml](https://github.com/Alvearie/alvearie-helm/blob/main/charts/ibm-fhir-server/templates/deployment.yaml#L159). This setting allows the stopping and deleting of jobs on all the hosts in the deployment.
```
Comment (lmsurpre) [inline docs/src/pages/guides/FHIRServerUsersGuide.md:1491]: wish there was a better way to keep these links up-to-date...but probably easiest for now just to make it right for this point-in-time
```suggestion
If you are running in a Kubernetes deployment, be sure to set the environment variable MY_POD_NAME to `metadata.name` as shown in [deployment.yaml](https://github.com/Alvearie/alvearie-helm/blob/main/charts/ibm-fhir-server/templates/deployment.yaml#L176). This setting allows the stopping and deleting of jobs on all the hosts in the deployment.
```

File: fhir-server-webapp/src/main/liberty/config/configDropins/disabled/postgres/bulkdata.xml [hunk 2]
```diff
@@ -32,6 +32,7 @@
         </application-bnd>
     </webApplication>
 
+    <variable name="BATCH_DB_HOSTNAME" defaultValue=""/>
```
[the remark this answers is not in the document]
Comment (lmsurpre) [reply fhir-server-webapp/src/main/liberty/config/configDropins/disabled/postgres/bulkdata.xml:35]: looks like our db2-cloud bulkdata.xml is missing a few variable declarations though...should we clean that one up too?

[the remark this answers is not in the document]
Comment (lmsurpre) [reply fhir-server-webapp/src/main/liberty/config/configDropins/disabled/postgres/bulkdata.xml:35]: specifically, that one is missing
* BATCH_DB_HOSTNAME
* BATCH_DB_APIKEY

File: fhir-server-webapp/src/main/liberty/config/configDropins/defaults/bulkdata.xml [hunk 3]
```diff
@@ -37,4 +37,10 @@
 
     <variable name="BATCH_DB_SCHEMA" defaultValue="FHIR_JBATCH"/>
     <databaseStore dataSourceRef="fhirbatchDS" id="BatchDatabaseStore" schema="${BATCH_DB_SCHEMA}" tablePrefix="" createTables="false"/>
+
+    <!--
+         The defaultHostName is set in the deployment.yaml https://github.com/Alvearie/alvearie-helm/blob/main/charts/ibm-fhir-server/templates/deployment.yaml#L159
+         and is subsequently used by the bulkdata configurations
+    -->
+    <variable name="defaultHostName" value="${env.MY_POD_NAME}"/>
```
Comment (lmsurpre) [inline fhir-server-webapp/src/main/liberty/config/configDropins/defaults/bulkdata.xml:45]: what happens if its not set?

File: docs/src/pages/guides/FHIRServerUsersGuide.md [hunk 4]
```diff
@@ -1410,7 +1410,7 @@ https://s3.appdomain.cloud/fhir-example/Patient_1.ndjson?X-Amz-Algorithm=AWS4-HM
 
 The presigned URL is valid for 86400 seconds (1 day).
 
-Note, the deletion of an a job is split into two phases, ACCEPTED (202) response and DELETED (204).  202 is returned until the operation is stopped or removed, and then 204.
+Note, the deletion of an a job is split into two phases, ACCEPTED (202) response and then 404 when the job is no longer available.
```
[the remark this answers is not in the document]
Comment (lmsurpre) [reply docs/src/pages/guides/FHIRServerUsersGuide.md:1413]: Assuming thats right, we could probably clear this up by covering all 3 cases in the doc:
```suggestion
To cancel a running job (or delete a completed job), issue an HTTP DELETE request to the polling URL.
* Invoking DELETE on a running job will return HTTP 202 (Accepted) and stop the job
* Invoking DELETE on a completed job will return HTTP 204 (Deleted) and delete the job
* Invoking DELETE on a deleted job will return a 404
```

Or if my assumption above is wrong, just let me know :-)

[the remark this answers is not in the document]
Comment (prb112) [reply docs/src/pages/guides/FHIRServerUsersGuide.md:1413]: Looks good

File: fhir-server-webapp/src/main/liberty/config/configDropins/disabled/postgres/bulkdata.xml (hunk 2 above)
[the remark this answers is not in the document]
Comment (prb112) [reply fhir-server-webapp/src/main/liberty/config/configDropins/disabled/postgres/bulkdata.xml:35]: Done

**Pull request merged**